src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/PackageActionBaseCommand.cs

```diff
@@ -293,8 +293,23 @@ protected override void EndProcessing()
 
         protected async Task CheckPackageManagementFormat()
         {
-            // check if Project has any packages installed
-            if (!(await Project.GetInstalledPackagesAsync(Token)).Any())
+            bool packagesConfigAndSupportsPackageReferences = false;
+
+            if (Project.ProjectStyle == ProjectModel.ProjectStyle.PackagesConfig)
```

Comment:
Why is this check needed? Only the packages config can be upgraded to packagereference?

Reply:
Yes, PC -> PR makes sense. Our logic will break if we start an upgrade here unnecessarily (that's the bug). If it's a new project, and the setting is PackageReference, we don't need to perform any upgrade logic.